test/cpsubsystem/AtomicLongProxyTest.js

```diff
@@ -0,0 +1,162 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { expect } = require('chai');
+const fs = require('fs');
+const Long = require('long');
+const RC = require('./../RC');
+const {
+    Client,
+    DistributedObjectDestroyedError
+} = require('../../');
+
+describe('AtomicLongProxyTest', function () {
+
+    this.timeout(30000);
+
+    let cluster;
+    let client;
+    let long;
+
+    function expectLong(expected, long) {
+        return expect(long.toString()).to.equal(Long.fromValue(expected).toString());
+    }
+
+    before(async function () {
+        cluster = await RC.createCluster(null, fs.readFileSync(__dirname + '/hazelcast_cpsubsystem.xml', 'utf8'))
+        await Promise.all([
+            RC.startMember(cluster.id),
+            RC.startMember(cluster.id),
+            RC.startMember(cluster.id)
+        ]);
+        client = await Client.newHazelcastClient({ clusterName: cluster.id });
+        long = await client.getAtomicLong('along');
+    });
+
+    afterEach(async function () {
+        // return to default value
+        await long.set(0);
+    });
+
+    after(async function () {
+        client.shutdown();
+        return RC.shutdownCluster(cluster.id);
+    });
+
+    it('should create AtomicLong with respect to given CP group', async function () {
+        const longInAnotherGroup = await client.getAtomicLong('along@mygroup');
+
+        const value1 = await longInAnotherGroup.incrementAndGet();
+        expectLong(1, value1);
+        // the following value has to be 0,
+        // as `long` belongs to the default CP group
+        const value2 = await long.get();
+        expectLong(0, value2);
+    });
+
+    it('destroy: should destroy AtomicLong', async function () {
+        const anotherLong = await client.getAtomicLong('another-long');
+        await anotherLong.destroy();
+
+        try {
+            await anotherLong.get();
+        } catch (err) {
+            expect(err).to.be.instanceOf(DistributedObjectDestroyedError);
+        }
+    });
+
+    it('get: should return 0 initially', async function () {
+        const value = await long.get();
+        expectLong(0, value);
+    });
+
+    it('addAndGet: should return added value', async function () {
+        const value = await long.addAndGet(33);
+        expectLong(33, value);
+    });
+
+    it('addAndGet: should add', async function () {
+        await long.addAndGet(33);
+        const value = await long.get();
+        expectLong(33, value);
+    });
+
+    it('getAndAdd: should return old value', async function () {
+        const value = await long.getAndAdd(123);
+        expectLong(0, value);
+    });
+
+    it('getAndAdd: should add', async function () {
+        await long.getAndAdd(123);
+        const value = await long.get();
+        return expectLong(123, value);
+    });
+
+    it('decrementAndGet: should decrement', async function () {
```

Comment:
Looking at the other tests, I see that this one does not have a counterpart in which you test the `get` value after a call to `decrementAndGet` (like `getAndAdd: should return old value` and `getAndAdd: should add`)

Reply:
That's because `decrementAndGet` returns the new value, not the old one like `getAndAdd`. So, we expect to get the new value in this very test.

Reply:
I initially thought that but after seeing `addAndGet: should return added value` and `addAndGet: should add`, I thought you may missed these two. 

But, I guess it is okay right now as they are using the `addAndGet` internally.

Reply:
> I initially thought that but after seeing `addAndGet: should return added value` and `addAndGet: should add`, I thought you may missed these two.

Those tests are a port of tests from the old AtomicLong test suite. But we should keep them in a good shape. Removed redundant test in 43dd117e433a4f1651ce35a4ab5a4aa9bdf7c240